server.js

```diff
@@ -10,16 +10,19 @@ const DATA_DIR = path.join(__dirname, 'data');
 
 app.use(express.json());
 
-const validateWalletAddress = (req, res) => {
+const validateWalletAddress = (req, res, next) => {
   if (!req.body.walletAddress) {
-    res.status(400).send({ message: 'Missing wallet address' });
-    return false;
+    return next({ code: 400, message: 'Missing wallet address' });
```

Comment:
```suggestion
    return next(Object.assign(new Error('Missing wallet address', { code: 400 })));
```
Always use real error instances. But feel free to extend them with extra properties.

Reply:
Added new class:

```js
class HttpError extends Error {
  constructor(message, code) {
    super(message);
    this.code = code;
  }
}
```

usage:

```js
  return next(new HttpError('Missing wallet address', 400));
```